unison-runtime/package.yaml

```diff
@@ -12,12 +13,28 @@ flags:
     manual: true
     default: false
 
+  # Run optimization assertion tests, make sure this runs with O2
+  optchecks:
+    manual: true
+    default: false
+
+  # Dumps core for debugging to unison-runtime/.stack-work/dist/<arch>/ghc-x.y.z/build/
+  dumpcore:
+    manual: true
+    default: false
+
 when:
   - condition: flag(arraychecks)
     cpp-options: -DARRAY_CHECK
   - condition: flag(stackchecks)
     cpp-options: -DSTACK_CHECK
-
+  - condition: flag(optchecks)
+    ghc-options: -O2
+    cpp-options: -DOPT_CHECK
+    dependencies:
+      - inspection-testing
+  - condition: flag(dumpcore)
+    ghc-options: -ddump-simpl -ddump-stg-final -ddump-to-file -dsuppress-coercions -dsuppress-idinfo -dsuppress-module-prefixes -ddump-str-signatures -ddump-simpl-stats # -dsuppress-type-applications -dsuppress-type-signatures
```

Comment:
I'm going to leave these commented out flags here because they're sometimes useful and it's annoying to go look them up every time I need them.